src/machine.rs

```diff
@@ -1379,6 +1379,42 @@ impl OlmMachine {
         })
     }
 
+    /// Get the stored room settings, such as the encryption algorithm or
+    /// whether to encrypt only for trusted devices.
+    ///
+    /// These settings can be modified via {@link #setRoomSettings}.
+    ///
+    /// # Returns
+    ///
+    /// `Promise<RoomSettings|undefined>`
+    #[wasm_bindgen(js_name = "getRoomSettings")]
+    pub async fn get_room_settings(
+        &self,
+        room_id: &identifiers::RoomId,
+    ) -> Result<JsValue, JsError> {
+        let result = self.inner.room_settings(&room_id.inner).await?;
+        Ok(result.map(|settings| RoomSettings::from(settings)).into())
+    }
+
+    /// Store encryption settings for the given room.
+    ///
+    /// This method checks if the new settings are "safe" -- ie, that they do
+    /// not represent a downgrade in encryption security from any previous
+    /// settings. Attempts to downgrade security will result in an error.
```

Comment:
I don't see these checks in this code. Are they in the code we call into?

Reply:
Yup.